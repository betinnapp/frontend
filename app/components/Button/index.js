/**
 *
 * Button
 *
 */

import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const StyledButton = styled.button`
  outline: none;
  background: ${({ secondary }) => (secondary && '#904ee2') || '#00dba1'};
  border: none;
  border-radius: 16px;
  width: 296px;
  height: 60px;
  text-transform: uppercase;
  color: #f4f4f4;
  font-size: 18px;
  font-weight: bold;
  transition: ease 0.3s;

  &:hover {
    filter: brightness(85%);
  }

  &:active {
    transform: translateY(2px);
  }

  &:disabled,
  &:disabled&:hover {
    background: #a3a3a3;
  }
`

function Button(props) {
  return (
    <StyledButton
      type={props.type}
      id={props.id}
      onClick={props.onClick}
      disabled={props.disabled}
      secondary={props.secondary}
    >
      {props.children}
    </StyledButton>
  )
}

Button.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  type: PropTypes.string,
  disabled: PropTypes.bool,
  secondary: PropTypes.bool,
}

Button.defualtProps = {
  type: 'button',
}

export default Button
