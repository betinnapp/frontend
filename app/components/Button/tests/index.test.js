/**
 *
 * Tests for Button
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from 'react-testing-library'

import Button from '../index'

describe('<Button />', () => {
  const localProps = {
    id: 'id123',
    onClick: () => {},
  }

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Button {...localProps}>Test</Button>)
    expect(firstChild).toMatchSnapshot()
  })
})
