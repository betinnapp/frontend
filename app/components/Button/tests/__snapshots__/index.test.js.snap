// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should render and match the snapshot 1`] = `
<button
  class="Button__StyledButton-ss41oc-0 eUuPsm"
  id="id123"
  type="button"
>
  Test
</button>
`;
